log.go

```diff
@@ -0,0 +1,37 @@
+package libhoney
+
+import (
+	"fmt"
+	"time"
+)
+
+// Logger is used to log extra info within the SDK detailing what's happening.
+// You can set a logger during initialization. If you leave it unititialized, no
+// logging will happen. If you set it to the DefaultLogger, you'll get
+// timestamped lines sent to STDOUT. Pass in your own implementation of the
+// interface to send it in to your own logger.
+type Logger interface {
+	// Log accepts the same msg, args style as fmt.Printf().
+	Log(msg string, args ...interface{})
```

Comment:
Can we make this method be `Printf(msg string, args ...interface{})` instead? That way you can pass an unmodified implementation of the standard library's `Logger` interface https://golang.org/pkg/log/#Logger.Printf .

Reply:
Ooh, excellent. Yes, absolutely.